### ISSUE TITLE
Control how content gets marked "Watched" after watching

### ISSUE THREAD

**Issue by ginjohn5snaps** [croneter/PlexKodiConnect#341]
Hello everyone,

I am having issues with the watched status of my content. If I for example watch an episode of a series in KODI with PlexKodiConnect, and stop the episode at the credits, then the episode will stay in a state where I can resume it. Sometimes it marks it watched, sometimes not. I guess that has to do with how many % of the content I have actually stopped at?

If it has not marked it watched, I will mark it watched in KODI, but the resumable state will remain. Sometimes I can start the episode again, and stop it right away, and it will remove the resumable state.

Sometimes when I tinkered to get the stuff looking right, it will be messed up after while - I guess when PlexKodiConnect resyncs with my Plex server. Also, I have Trakt running in Plex if that matters.

Any clue what is going on? How do you guys manage this?

**Comment by croneter** [comment]
PKC checks whether you've watchd >=90% of a video. If yes, PKC tells Kodi to mark the video as watched. For some reason, this does not always work. 
* Your PMS should always have the "correct" viewstate, so "fully watched" after 90%
* During full syncs (usually every 60min), PKC will pick up on the correct playstate and mark the video as fully watched

No idea how Trakt is playing in here. Have you tried deactivating it and checking whether the issues persist? 

I'll make some changes to the newest beta version of PKC hopefully solving this

**Comment by ginjohn5snaps** [comment]
Thank you for your reply.

Actually I did try disabling trakt yesterday, but it seems to make no difference. So I am glad to hear you will be trying out changes in next beta. Thank you for your good work.

**Comment by croneter** [comment]
Beta is being pushed right now, so in a couple of minutes you should be able to update to 1.8.14. Please play with it and report back 😉

**Comment by croneter** [comment]
Any improvement?

**Comment by ginjohn5snaps** [comment]
Did not have time to check it out just yet. Will get to it soon :-) I am using the PKC off the repo. Can I get the beta there?

**Comment by croneter** [comment]
No worries about the repo, this change is now also in the stable version

**Comment by Traxxus** [comment]
Different person, same chronic problem. On .15, they seem to more consistently mark as watched when played all the way through now at least, (3 for 3), but stopping a little early still leaves it at 98-99%.

**Comment by Traxxus** [comment]
Watched 2 60 (44) minute episodes today, did not update as watched even though watched all the way through, showed up as 99. The 3 that did update yesterday were 30 (21) minute episodes.

**Comment by croneter** [comment]
Darn!

Did you try doing that while tract was **deactivated**? No need to watch the entire movie/episode, just jump to the last couple of seconds

**Comment by Traxxus** [comment]
Yeah, removed Trakt from Plex. Ran through a few, 3 episodes marked as watched, one at 99. The last 2, they did mark as watched initially, but reverted to 99% after a couple seconds which I found interesting.  

This reminded me of a similar problem I had in Emby, though it didn't have anything to do with percentage, just watched episodes immediately reverting to unwatched, such as here. https://emby.media/community/index.php?/topic/24054-watched-episodes-keep-getting-marked-unwatched/

Just throwing that out there since it was such similar behavior.
